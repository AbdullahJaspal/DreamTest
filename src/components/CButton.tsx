import React from 'react';
import {StyleSheet, TouchableOpacity, StyleProp, ViewStyle} from 'react-native';
import {BORDER, COLOR, SPACING, TEXT} from '../configs/styles';
import CText from '../components/CText';

interface CButtonProps {
  children?: React.ReactNode;
  activeOpacity?: number;
  onPress?: () => void;
  style?: StyleProp<ViewStyle>;
  lable?: string;
  colorLabel?: string;
  sizeLabel?: number;
  width?: number;
  height?: number;
  borderRadius?: number;
  backgroundColor?: string;
  onLongPress?: () => void;
}

const CButton: React.FC<CButtonProps> = ({
  children,
  activeOpacity = 0.8,
  onPress,
  style,
  lable,
  colorLabel = COLOR.WHITE,
  sizeLabel,
  width,
  height,
  borderRadius = BORDER.SMALL,
  backgroundColor = COLOR.DANGER2,
  onLongPress,
}) => {
  const styles = StyleSheet.create({
    container: {
      borderRadius,
      padding: SPACING.S2,
      backgroundColor,
      justifyContent: 'center',
      alignItems: 'center',
      width,
      height,
    },
  });

  return (
    <TouchableOpacity
      style={[styles.container, style]}
      activeOpacity={activeOpacity}
      onPress={onPress}
      onLongPress={onLongPress}>
      {lable ? (
        <CText text={TEXT.REGULAR} color={colorLabel} fontSize={sizeLabel}>
          {lable}
        </CText>
      ) : (
        children
      )}
    </TouchableOpacity>
  );
};

export default CButton;
