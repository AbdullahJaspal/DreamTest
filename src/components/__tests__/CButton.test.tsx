import 'react-native';
import React from 'react';

import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import CButton from '../CButton';

it('Icon renders correctly', () => {
  const tree = renderer.create(<CButton></CButton>).toJSON();

  expect(tree).toMatchSnapshot();
});
