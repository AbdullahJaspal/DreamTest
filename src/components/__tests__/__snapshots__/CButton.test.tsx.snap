// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FE2C55",
      "borderRadius": 4,
      "height": undefined,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
      "width": undefined,
    }
  }
/>
`;
